lib/pbench/server/api/resources/query_apis/datasets/namespace_and_rows.py

```diff
@@ -67,7 +67,7 @@ def assemble(self, params: ApiParams, context: CONTEXT) -> JSON:
         }
         """
         dataset: Dataset = context["dataset"]
-        document = self.ES_INTERNAL_INDEX_NAMES[params.uri["dataset_view"]]
+        document = self.ES_INTERNAL_INDEX_NAMES[params.uri.get("dataset_view")]
```

Comment:
Failure here is unlikely ("impossible"?) because `dataset_view` is validated as a required keyword: the `dataset_view` URI parameter has to be present. That makes the change from direct indexing to `.get` inefficient and unnecessary.

But more than that, if there was somehow a failure the old way would throw a key error *here* which would be caught in the dispatcher with a log message referencing the key `dataset_view` while the new way would raise an exception complaining about trying to index `None` below at `document["index"]`... which isn't nearly as helpful.

Although it's probably not a serious issue, I think you should revert this change.